visualization/app/codeCharta/util/deltaGenerator.ts

```diff
@@ -81,6 +83,21 @@ export class DeltaGenerator {
 		}
 	}
 
+	private static compareCommonAttributes(nodeA: CodeMapNode, nodeB: CodeMapNode): { differenceExists: boolean } {
+		const nodesAreComparable = nodeA.type === NodeType.FILE && nodeB.type === NodeType.FILE && nodeA.attributes && nodeB.attributes
+
+		if (!nodesAreComparable) {
+			return { differenceExists: false }
+		}
+
+		const differenceExists = Object.keys(nodeA.attributes).some(attribute => {
+			const bHasAttribute = nodeB.attributes[attribute] !== undefined
+			const valuesDiffer = nodeB.attributes[attribute] !== nodeA.attributes[attribute]
+			return bHasAttribute && valuesDiffer
+		})
+		return { differenceExists }
+	}
+
```

Comment:
We should consider `A` containing a metric and `B` does not to be `changed`. It might be wrong but we do visualize a changed metric in the overview. As such, it's consistent to consider the metric as "changed".

The logic itself would work but it's a significant overhead as we already do the heavy lifting in `getDeltaAttributeList()`.

Instead we can simplify this by reworking the getDeltaAttributeList. We should change the name and the return type.

```js
let changed = false;
for (const key of Object.keys(comparisonAttribute)) {
	if (referenceAttribute[key]) {
		const difference = comparisonAttribute[key] - referenceAttribute[key]
		deltaAttribute[key] = difference
		if (difference !== 0) {
			changed = true
		}
	} else {
		changed = true
		deltaAttribute[key] = comparisonAttribute[key]
	}
}
for (const key of Object.keys(referenceAttribute)) {
	if (!comparisonAttribute[key]) {
		deltaAttribute[key] = -referenceAttribute[key]
		changed = true
	}
}
```

Reply:
Resolved by commit [d19757](https://github.com/MaibornWolff/codecharta/pull/2890/commits/9d19757baf9002ea08b9ba27766a8eb297719a60)